services/navet-ms/lambdas/pollUser.js

```diff
@@ -1,78 +1,106 @@
-      city: navetData.Adresser.Folkbokforingsadress.Postort,
+      street: navetUser.Adresser.Folkbokforingsadress.Utdelningsadress2,
+      postalCode: navetUser.Adresser.Folkbokforingsadress.PostNr,
+      city: navetUser.Adresser.Folkbokforingsadress.Postort,
     },
-    civilStatus: navetData.Civilstand.CivilstandKod,
+    civilStatus: navetUser.Civilstand.CivilstandKod,
   };
-  return userObj;
+
+  return eventDetail;
 }
 
-async function requestNavetUser(payload, params) {
-  let err, navetClient, navetUser, parsedData;
+async function requestNavetUser(personalNumber) {
+  const ssmParams = await NAVET_PARAMS;
+  ssmParams.personalNumber = personalNumber;
 
-  // Create HTTP client
-  [err, navetClient] = await to(client(params));
-  if (!navetClient) throwError(503);
+  const [getNavetClientError, navetRequestClient] = await to(getNavetRequestClient(ssmParams));
+  if (getNavetClientError) {
+    throw getNavetClientError;
+  }
 
-  // Get user from Navet API
-  [err, navetUser] = await to(
-    request.call(navetClient, 'post', params.personpostXmlEndpoint, payload)
+  const navetPostRequestXmlPayload = parseXml(ssmParams);
+
+  const [postNavetClientError, navetUser] = await to(
+    request.call(
+      navetRequestClient,
+      'post',
+      ssmParams.personpostXmlEndpoint,
+      navetPostRequestXmlPayload
+    )
   );
-  if (err) {
-    // Parse Error Message from XML.
-    const [, parsedErrorMessage] = await to(parseErrorMessageFromXML(err.response.data));
-    throwError(err.response.status, parsedErrorMessage || null);
+  if (postNavetClientError) {
+    if (postNavetClientError.response) {
+      const [, navetResponseXmlErrorMessage] = await to(
+        parseErrorMessageFromXML(postNavetClientError.response.data)
+      );
+
+      throw navetResponseXmlErrorMessage;
+    }
+
+    throw postNavetClientError;
   }
 
-  // Parse XML data to JSON
-  [err, parsedData] = await to(parseJSON(navetUser.data));
-  if (err) throwError(500);
+  return await getNavetUserPersonPostAsJson(navetUser);
+}
 
-  const { Folkbokforingspost } = parsedData;
+async function getNavetUserPersonPostAsJson(navetUser) {
+  const [parseJsonError, navetUserJson] = await to(parseJSON(navetUser.data));
+  if (parseJsonError) {
+    throw parseJsonError;
+  }
+
+  const { Folkbokforingspost } = navetUserJson;
+  if (Folkbokforingspost.Personpost === undefined) {
+    throw new InternalServerError();
+  }
 
-  // Abort if user data is confidential
-  if (
-    Folkbokforingspost.Sekretessmarkering === 'J' ||
-    Folkbokforingspost.SkyddadFolkbokforing === 'J'
-  ) {
-    throw new Error('User is not allowed to use the service.');
+  if (isUserConfidential(navetUserJson)) {
+    throw new UnauthorizedError();
   }
 
-  // Collect the user data from response
-  const navetUserData = Folkbokforingspost.Personpost;
-  if (navetUserData === undefined) throwError(500);
+  return Folkbokforingspost.Personpost;
+}
+
+function isUserConfidential(navetUser) {
```

Comment:
isNavetPersonCondifential seems more correct since we don't want to this to be missunderstud as a aws user.